### DISCUSSION TITLE
Problems with PWMERGE_DGR

### DISCUSSION THREAD

**Discussion by danboid** [ltsp/ltsp#668] [Q&A]
My issue is that our we use LDAP for auth on our LTSP server and all the users are in GID 1001 (isdads-user). I have no access to the LDAP server and so I've had to work around its existing config.

I created the required group in the source VM image used to create our LTSP image and added a local user to that group but LTSP wasn't transferring the group the group to the image so every time a user opened a terminal they would see the error:

```
groups: cannot find name for group ID 1001
```

I asked @alkisg about this in IRC (and on the LTSP discussions page) and he said the solution was to add:

```
PWMERGE_DGR="isdads-user"
```

to the [clients] section of /etc/ltsp/ltsp.conf on the LTSP server but this doesn't work by itself. The cleanest workaround I have found so far is to edit line 34 (the line that calls pwmerge) in `/usr/share/ltsp/server/image/55-cleanup.sh` to look like:

```
re "$_LTSP_DIR/client/login/pwmerge" --ltsp --quiet --dgr=isdads-user \
```

Without adding `--dgr=isdads-user`, `pwmerge` removes this group from `/etc/group`, even though it has a local user as a member.

To fix the missing group error we had to both edit 55-cleanup.sh and add PWMERGE_DGR to ltsp.conf. This has got rid of the groups error but isn't a fix I'm happy with because 55-cleanup.sh will need to be patched every time LTSP gets updated, unless we use `apt-mark hold ltsp` to stop ltsp getting updated.

**Comment by alkisg** [comment]
Try under [common].

**Comment by danboid** [comment]
I'll try it it under [common] next week.

**Reply by danboid** [reply]
Yes, adding PWMERGE_DGR to the [common] section instead has fixed the error.

The LTSP man page should document which section each directive belongs under.

Thanks Alkis!

**Reply by danboid** [reply]
Adding PWMERGE_DGR to [common] doesn't quite work. On the first boot or two that I tried after adding this I didn't get the cannot find group name error but for some reason it stopped working shortly after so this doesn't work in all situations.

**Reply by danboid** [reply]
`ltsp image` does spew a bunch of errors because it can't remove a bunch of temp dirs every time I rebuild the image so I wonder if thats linked?

**Reply by danboid** [reply]
I've not forgot about my "Notes on setting up an advanced LTSP server" guide - that is coming shortly. I just wanted to test it for a month or so first. Unfortunately I think I'll have to add my dirty hack for fixing this unless we can work out why PWMERGE_DGR isn't working for me.

**Reply by danboid** [reply]
I no longer have the `ltsp image` temp dir removal problem and I can confirm that wasn't the cause of `PWMERGE_DGR` not working from ltsp.conf reliably.

My guide is getting uploaded this week. I will add a link to it on the wiki etc.
